packages/ra-ui-materialui/src/input/NumberInput.js

```diff
@@ -73,7 +73,7 @@ export class NumberInput extends Component {
                 margin="normal"
                 error={!!(touched && error)}
                 helperText={touched && error}
-                step={step}
+                inputProps={{step}}
```

Comment:
No, it should be in `inputProps`. See the material-ui doc: https://material-ui-next.com/api/text-field/

Reply:
> inputProps => Properties applied to the native input element.

So here, we set the `step` attribute which is supported by native inputs